test/unified_format.py

```diff
@@ -0,0 +1,973 @@
+    def parse_collection_or_database_options(options):
+        return parse_collection_options(options)
+
+    @staticmethod
+    def parse_bulk_write_result(result):
+        upserted_ids = {str(int_idx): result.upserted_ids[int_idx]
+                        for int_idx in result.upserted_ids}
+        return {
+            'deletedCount': result.deleted_count,
+            'insertedCount': result.inserted_count,
+            'matchedCount': result.matched_count,
+            'modifiedCount': result.modified_count,
+            'upsertedCount': result.upserted_count,
+            'upsertedIds': upserted_ids}
+
+    @staticmethod
+    def parse_bulk_write_error_result(error):
+        write_result = BulkWriteResult(error.details, True)
+        return SpecTestUtil.parse_bulk_write_result(write_result)
+
+
+class EventListenerUtil(CommandListener):
+    def __init__(self, observe_events, ignore_commands):
+        self._event_types = set(observe_events)
+        self._ignore_commands = set(ignore_commands)
+        self._ignore_commands.add('configureFailPoint')
+        self.results = []
+
+    def _observe_event(self, event):
+        if event.command_name not in self._ignore_commands:
+            self.results.append(event)
+
+    def started(self, event):
+        if 'commandStartedEvent' in self._event_types:
+            self._observe_event(event)
+
+    def succeeded(self, event):
+        if 'commandSucceededEvent' in self._event_types:
+            self._observe_event(event)
+
+    def failed(self, event):
+        if 'commandFailedEvent' in self._event_types:
+            self._observe_event(event)
+
+
+class EntityMapUtil(object):
+    """Utility class that implements an entity map as per the unified
+    test format specification."""
+    def __init__(self, test_class):
+        self._entities = {}
+        self._listeners = {}
+        self._session_lsids = {}
+        self._test_class = test_class
+
+    def __getitem__(self, item):
+        try:
+            return self._entities[item]
+        except KeyError:
+            self._test_class.fail('Could not find entity named %s in map' % (
+                item,))
+
+    def __setitem__(self, key, value):
+        if not isinstance(key, text_type):
+            self._test_class.fail(
+                'Expected entity name of type str, got %s' % (type(key)))
+
+        if key in self._entities:
+            self._test_class.fail('Entity named %s already in map' % (key,))
+
+        self._entities[key] = value
+
+    def _create_entity(self, entity_spec):
+        if len(entity_spec) != 1:
+            self._test_class.fail(
+                "Entity spec %s did not contain exactly one top-level key" % (
+                    entity_spec,))
+
+        entity_type, spec = next(iteritems(entity_spec))
+        if entity_type == 'client':
+            # TODO
+            # Add logic to respect the following fields
+            # - uriOptions
+            # - useMultipleMongoses
```

Comment:
Can you update this comment to include the jira ticket which will implement these features (and create a new ticket if it doesn't already exist)?

Reply:
These opts are now implemented so this comment is moot.